ui/plugins/controller.js

```diff
@@ -390,6 +390,13 @@ treeherder.controller('PluginCtrl', [
         $scope.backfillButtonTitle = function() {
             var title = "";
 
+            // Ensure currentRepo is available on initial page load
+            if (!$scope.currentRepo) {
```

Comment:
We shouldn't change the global state here. Let's just say something like this:

```js
if (!$scope.currentRepo) {
  // still loading
  return undefined;
}
```

That way we just won't bind the property while we're still loading, which is what we want (you'll also want to revert your change below).